packages/dapp-svelte-wallet/api/src/lib-wallet.js

```diff
@@ -72,7 +73,44 @@ export function makeWallet({
   myAddressNameAdmin,
   pursesStateChangeHandler = noActionStateChangeHandler,
   inboxStateChangeHandler = noActionStateChangeHandler,
+  localTimerService,
 }) {
+  /**
+   * The current timestamp, in milliseconds (if it is known).
+   *
+   * @type {number | undefined}
+   */
+  let nowMs;
+
+  // Subscribe to the timer service to update our stamp.
+  if (localTimerService) {
+    observeNotifier(E(localTimerService).makeNotifier(0n, 1000n), {
+      updateState(bigStamp) {
+        // We need to convert the bigint to a number (for easy Javascript date math).
+        nowMs = parseInt(`${bigStamp}`, 10);
+      },
+    });
+  }
+
+  let lastSequence = 0;
+  const addMeta = record => {
+    const { meta: oldMeta = {} } = record;
+    const meta = { ...oldMeta };
+    if (!meta.sequence) {
+      // Add a sequence number to the record.
+      lastSequence += 1;
```

Comment:
What's the purpose of the sequence?

Reply:
Rather, why do we need this if we have the timestamps, because they're not granular enough?

Reply:
Sequence is guaranteed to be ordered and unique, even if the wallet instantiator didn't supply a timerService.

Timestamps are just informational for human consumption (after formatting).  And yes, they may not be granular enough because the timer updates only happen every second.

Reply:
These are the same questions I had. Can we add the answers as comments?